daprdocs/content/en/java-sdk-docs/_index.md

```diff
@@ -0,0 +1,229 @@
+}
+```
+
+## Building blocks
+
+The Java SDK allows you to interface with all of the [Dapr building blocks]({{< ref building-blocks >}}).
+
+### Invoke a service
+
+```java
+import io.dapr.client.DaprClient;
+import io.dapr.client.DaprClientBuilder;
+
+try (DaprClient client = (new DaprClientBuilder()).build()) {
+  // invoke a 'GET' method (HTTP) skipping serialization: \say with a Mono<byte[]> return type
+  // for gRPC set HttpExtension.NONE parameters below
+  response = client.invokeMethod(SERVICE_TO_INVOKE, METHOD_TO_INVOKE, "{\"name\":\"World!\"}", HttpExtension.GET, byte[].class).block();
+
+  // invoke a 'POST' method (HTTP) skipping serialization: to \say with a Mono<byte[]> return type     
+  response = client.invokeMethod(SERVICE_TO_INVOKE, METHOD_TO_INVOKE, "{\"id\":\"100\", \"FirstName\":\"Value\", \"LastName\":\"Value\"}", HttpExtension.POST, byte[].class).block();
+
+  System.out.println(new String(response));
+
+  // invoke a 'POST' method (HTTP) with serialization: \employees with a Mono<Employee> return type      
+  Employee newEmployee = new Employee("Nigel", "Guitarist");
+  Employee employeeResponse = client.invokeMethod(SERVICE_TO_INVOKE, "employees", newEmployee, HttpExtension.POST, Employee.class).block();
+}
+```
+
+- For a full guide on service invocation visit [How-To: Invoke a service]({{< ref howto-invoke-discover-services.md >}}).
+- Visit [Java SDK examples](https://github.com/dapr/java-sdk/tree/master/examples/src/main/java/io/dapr/examples/invoke) for code samples and instructions to try out service invocation
+
+### Save & get application state
+
+```java
+import io.dapr.client.DaprClient;
+import io.dapr.client.DaprClientBuilder;
+import io.dapr.client.domain.State;
+import reactor.core.publisher.Mono;
+
+try (DaprClient client = (new DaprClientBuilder()).build()) {
+  // Save state
+  client.saveState(STATE_STORE_NAME, FIRST_KEY_NAME, myClass).block();
+
+  // Get state
+  Mono<State<MyClass>> retrievedMessageMono = client.getState(STATE_STORE_NAME, FIRST_KEY_NAME, MyClass.class);
+
+  // Delete state
+  client.deleteState(STATE_STORE_NAME, FIRST_KEY_NAME).block();
+}
+```
+
+- For a full list of state operations visit [How-To: Get & save state]({{< ref howto-get-save-state.md >}}).
+- Visit [Java SDK examples](https://github.com/dapr/java-sdk/tree/master/examples/src/main/java/io/dapr/examples/state) for code samples and instructions to try out state management
+
+### Publish & subscribe to messages
+
+##### Publish messages
+
+```java
+import io.dapr.client.DaprClient;
+import io.dapr.client.DaprClientBuilder;
+import io.dapr.client.domain.Metadata;
+import static java.util.Collections.singletonMap;
+
+try (DaprClient client = (new DaprClientBuilder()).build()) {
+  client.publishEvent(PUBSUB_NAME,TOPIC_NAME,message,singletonMap(Metadata.TTL_IN_SECONDS, MESSAGE_TTL_IN_SECONDS)).block();
+}
+```
+
+##### Subscribe to messages
+import com.fasterxml.jackson.databind.ObjectMapper;
```

Comment:
@paulyuk I think you want these inside the java code block.

Reply:
Let's use new new PR 2 with right branch mapping.  thx fixed issue above.
https://github.com/greenie-msft/java-sdk/pull/2